src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -826,6 +861,15 @@ Copyright (c) .NET Foundation. All rights reserved.
                                             Condition="'%(_NoneWithTargetPath.CopyToPublishDirectory)'=='PreserveNewest'"/>
     </ItemGroup>
 
+    <ItemGroup Condition="'$(PublishSingleFile)' == 'true' and '$(SelfContained)' == 'true' and $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0')) and Exists('$(SingleFileHostIntermediatePath)')">
```

Comment:
As with the `_CreateSingleFileHost` condition, the condition for the target framework should be expressed in terms of the `TargetFrameworkIdentifier` and `TargetFrameworkVersion` properties.  It might make sense to do something like consolidate the logic to refer to a single property in both places.